File: app/src/main/java/com/xuvjso/nfmovies/Utils/AESUtil.java
package com.xuvjso.nfmovies.Utils;

import android.util.Base64;

import javax.crypto.Cipher;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

public class AESUtil {

    public static String decryptData(String text, String key, String iv)throws Exception{

        byte[] encryted_bytes = Base64.decode(text, Base64.DEFAULT);

        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS7Padding");
        byte[] static_key = key.getBytes();

        SecretKeySpec keySpec = new SecretKeySpec(static_key, "AES");
        IvParameterSpec ivSpec = new IvParameterSpec(iv.getBytes());
        cipher.init(Cipher.DECRYPT_MODE, keySpec, ivSpec);

        byte[] decrypted = cipher.doFinal(encryted_bytes);
        String result = new String(decrypted);

        return result;
    }
}

File: app/src/main/java/com/futuretech/nfmovies/Utils/AESUtil.kt
package com.futuretech.nfmovies.Utils

import android.util.Base64

import javax.crypto.Cipher
import javax.crypto.spec.IvParameterSpec
import javax.crypto.spec.SecretKeySpec

object AESUtil {

    @Throws(Exception::class)
    fun decryptData(text: String, key: String, iv: String): String {

        val encryted_bytes = Base64.decode(text, Base64.DEFAULT)

        val cipher = Cipher.getInstance("AES/CBC/PKCS7Padding")
        val static_key = key.toByteArray()

        val keySpec = SecretKeySpec(static_key, "AES")
        val ivSpec = IvParameterSpec(iv.toByteArray())
        cipher.init(Cipher.DECRYPT_MODE, keySpec, ivSpec)

        val decrypted = cipher.doFinal(encryted_bytes)

        return String(decrypted)
    }
}